sdks/python/src/opik/integrations/sagemaker/auth.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+import httpx
+
+from typing import Any
+
+import opik.hooks
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _in_aws_sagemaker() -> bool:
+    return os.getenv("AWS_PARTNER_APP_AUTH") is not None
+
+
+class SagemakerAuth(httpx.Auth):
+    def __init__(self, auth_provider: Any) -> None:
+        self.auth_provider = auth_provider
+
+    def auth_flow(self, request):  # type: ignore
+        if not _in_aws_sagemaker():
+            yield request
+
+        url, signed_headers = self.auth_provider.get_signed_request(
+            request.url, request.method, request.headers, request.content
+        )
+
+        request.url = url
```

Comment:
The httpx' url is not string type. Can you try following code before assign it to request?
```
url = httpx.URL(url_str)
```